### ISSUE TITLE
Pads for UART

### ISSUE THREAD

**Issue by Yoda-BZH** [xfarrow/tapo-camera#4]
Have you identified the UART pads ?

Can you provide a schema with the identified pads to connect the UART to ?

Thank you

**Comment by ferrum** [comment]
Hi! The UART pads should be here https://drmnsamoliu.github.io/assets/img/UART.JPG

**Comment by Yoda-BZH** [comment]
This is for the C200, is it supposed to be valid for the C210 too ?

**Comment by ferrum** [comment]
@Yoda-BZH yes, at least in my C210 the pads are exactly in this position. If you can post a picture of your motherboard we could give a look together :)

**Comment by Yoda-BZH** [comment]
Bottom:
![2024-12-09_10-27](https://github.com/user-attachments/assets/59c8818c-90db-4591-bef6-b9205753fbf2)

Top:
![2024-12-09_10-29](https://github.com/user-attachments/assets/0a78ebf0-7ece-470f-914a-1639d3041a7c)

I think I identified it, noted 1 to 4. But they're not clearly identified.

**Comment by ferrum** [comment]
Oh, seems like the motherboard schematics is different from mine as well. I am not an expert, but I also think those you outlined are UART pads. However, the lack of clear identification labels makes the process harder than expected because positioning cables incorrectly could damage the camera. I am not able to provide any further information at this moment, but I will try to investigate the matter.

**Comment by klimeryk** [comment]
I have the same board as @Yoda-BZH, it seems. It's marked as `C210(EU) Ver. 2.20` (as seen also [on the photo above](https://github.com/xfarrow/tapo-c210/issues/4#issuecomment-2527390057)). It's using the SSC333 chipset, instead of the `SSC335` that I think your version has, @xfarrow.

I've found this https://www.youtube.com/watch?v=egNNp8ImRY0 video where someone went through the trouble of debugging these pins and documenting it (thank you!). So, using the numbers from the photo above, looks like the designation is:
 - 1: Tx
 - 2: Ground
 - 3: VCC
 - 4: Rx

3 and 4 are my guess, because it's not clearly shown on the video. It looks like they are trying pin 4 first for the receive pin and it works for them.

Additional thing worth noting from the video is that the `root` password for that particular board was `slplzgjipc`. Not sure if it's shared across the version of the camera, version of the board or if it's specific to that particular instance, but making a note here for easy future reference. Worst case scenario, looks like it follows an easy to crack format - the video includes a quick note about that too.

Also, slightly related issue in OpenIPC: https://github.com/OpenIPC/builder/issues/2. It's for the C110 camera, but it uses SSC333 as well, so making a note of it here in case some issues (like the WIFI chip) are similar.

**Comment by ferrum** [comment]
@klimeryk Thank you for your precious contribution!

**Comment by hjhjdev** [comment]
I have also received ssc333 with same marking which @Yoda-BZH  had;
Also, I was able to get UART connection with bridge in picure(red box).

![IMG_2322](https://github.com/user-attachments/assets/86a41275-df64-4c39-97b3-cb98b8d8d81f)

**Comment by Sandr0x00** [comment]
> Also, slightly related issue in OpenIPC: [OpenIPC/builder#2](https://github.com/OpenIPC/builder/issues/2). It's for the C110 camera, but it uses SSC333 as well, so making a note of it here in case some issues (like the WIFI chip) are similar.

The WiFi chip for `C210(EU) Ver. 2.20` is atbm6012b (https://github.com/OpenIPC/firmware/blob/134387dbf7989fa6f2f718fb6b7318f86068b92e/general/overlay/etc/wireless/usb#L212). the original firmware is lying in the UART log. (it's also possible to connect to the UART on the Chip directly based on https://linux-chenxing.org/infinity6/ssc333_pb_v01.pdf as done here: https://youtu.be/F-G-7-qo7Xg?si=bphXcZ7Ef69AD-AG&t=652) 
Installing OpenIPC with that WiFI Chip works like a charm. I also didn't need to set GPIO16 (even though the original firmware does 🤷 )

**Comment by verbage** [comment]
> I have also received ssc333 with same marking which [@Yoda-BZH](https://github.com/Yoda-BZH) had; Also, I was able to get UART connection with bridge in picure(red box).
> 
> ![IMG_2322](https://github.com/user-attachments/assets/86a41275-df64-4c39-97b3-cb98b8d8d81f)

@hjhjdev, to be 100% clear before I "operate" on my C210, are you saying that bridging the two micro leads in the red box basically enables pad #1, which is TX?  So the left micro lead in the red box is connected to the TX pin #23 on the SSC333?  I have not yet pulled the pcb from my camera and there is not a lot of working room without doing that, but yeah, with a simple multimeter continuity test, that seems to be the case.  But until I pull the pcb for better working access, I cannot see anything definitively.

**Comment by verbage** [comment]
Well, I pulled the pcb and had great expectations for micro-soldering, but no go for me.  Not only did I not bridge the gap that @hjhjdev pointed out, I somehow lost the R518 micro resistor just above it--ouch!  This said, in the meantime, before the micro-soldering started, I made very nice scans of the pcb.

I have the US version of the C210, and on the box, it is identified as the C210P2 with hardware version US/2.26.  It also has a sticker on the bottom side of the pcb with the same C210P2 US/2.26 labeling.

From cursory comparisons, the pcb seems pretty much identical to the C210(EU)/2.20 pcb that @Yoda-BZH provided photographs of above (same layout, same SSC333 SOC and other components, etc.).  Nonetheless, I include the scans of the C210P2 US/2.26 pcb here for posterity in case somebody with the same model (and better micro-soldering skills!) needs a jumping off point in the future.

![Image](https://github.com/user-attachments/assets/4cdc965b-d88c-4f23-98a5-19076af932d8)

![Image](https://github.com/user-attachments/assets/fe65f851-27e1-4a7f-8143-fa82c6f571b6)

**Comment by verbage** [comment]
Regarding the R518 micro resistor that I lost, I'm wondering if just bridging the gap with solder would be a solution?  From the scans, this R518 resistor seems to bridge lines 24 and 25 of the SSC333.  Those are the GND_EFUSE and PM_GPIO4 lines, respectively, so maybe not so critical?  I can always start it up without that resistor to see if it makes a difference.  Whatever the case, if I could not bridge the trace for the gap in the TX line, I'm not really confident I could fix the missing resistor issue either.

I think I'll make one more attempt to get UART TX/RX/GND leads soldered in tomorrow.  Those seem more reasonable than trying to bridge the trace! There's a huge difference with 2.54mm/0.1" pitch like Arduino-style, which is what I am used to.

Maybe I'll have to wait till somebody devises a no-tools MicroSD auto-recovery boot method--would be awesome to do that!

**Comment by verbage** [comment]
So, just a brief update here.  My C210P2 seems to work fine without the R518 micro-resistor.  In the meantime, I went back for a 2nd attempt at micro-soldering, and woo-hoo, I now have a UART connection.

My boot log looks very similar up to a certain point to the one pointed out by @klimeryk a few posts, back, i.e., the [one by cjdell for a Tapo C110, which uses the same SSC333 SOC](https://gist.github.com/cjdell/c48f95d9910f2e23fb0ee7dfa6a41d34).  But the significant difference is whereas cjdell's C110 boot log continues after the "## Starting application at 0x220A0000 ..." message, I get no more console output despite the camera starting up fine and working.

I had boneheadedly updated my C210P2 to the latest firmware when it arrived, and now I am kicking myself for that because I realize it might have tightened things up with the boot, and for that reason, I might not see more.  I have tried to downgrade the firmware via MicroSD card, but no luck with that at this point.  My C210P2 actually came as a 2-pack so I still have an untouched one with original firmware.

I have also more closely compared scans of my C210P2 US/2.26 pcb with @Yoda-BZH 's pics of the C210(EU)/2.20 pcb.  They seem to be 100% identical component-wise, and even 100% identical with respect to all the numbering/silk-screening on the board (except for obvious things like date/batch codes, which are expected to be different ).  The labeling on the base of my beige camera case lists it as "C210" only--with no "P2" extension.  And the main printing on the box is also just "C210".  The only places where the "P2" part is added are on the unique, printed barcode on the outside of the box and the sticker on the pcb.  Given this, I think there is no meaningful difference between the "C210" and "C210P2"--maybe just different norms for wifi between different global zones, and thus, maybe slightly different firmwares?  In fact, I'm thinking the "P2" might simply mean "Pack of 2", which is seen in some advertising.  In sum, they seem identical.  So I suspect one could cross-flash C210 firmware onto the C210P2, and vice versa.
